File: frontend/src/api/course.ts
import request from '@/utils/request'

// 课程查询参数
export interface CourseQueryParams {
  current?: number
  size?: number
  keyword?: string
  teacherId?: number
  classId?: number
  status?: number
  sortOrder?: 'asc' | 'desc'
}

// 课程 DTO
export interface CourseDTO {
  id?: number
  courseName: string
  teacherId?: number
  classId?: number
  credit?: number
  courseHours?: number
  description?: string
  coverUrl?: string
  status?: number
  startTime?: string
  endTime?: string
}

// 课程 VO
export interface CourseVO {
  id: number
  courseName: string
  teacherId: number
  teacherName?: string
  classId: number
  className?: string
  credit?: number
  courseHours?: number
  description?: string
  coverUrl?: string
  status: number
  startTime?: string
  endTime?: string
  createTime?: string
  createdAt?: string
  updatedAt?: string
  studentCount?: number
  assignmentCount?: number
  activeAssignmentCount?: number
  completionRate?: number
}

// 分页结果
export interface PageResult<T> {
  records: T[]
  total: number
  pages: number
  current: number
  size: number
}

// 获取课程列表
export const getCourseList = (params: CourseQueryParams) => {
  return request.get<PageResult<CourseVO>>('/course/list', { params })
}

// 新增课程
export const addCourse = (data: CourseDTO) => {
  return request.post('/course/add', data)
}

// 编辑课程
export const updateCourse = (data: CourseDTO) => {
  return request.put('/course/update', data)
}

// 删除课程
export const deleteCourse = (id: number) => {
  return request.delete(`/course/${id}`)
}

// 获取课程详情
export const getCourseById = (id: number) => {
  return request.get<CourseVO>(`/course/${id}`)
}

export const uploadCourseCover = (data: FormData) => {
  return request.post<string>('/course/cover', data, {
    headers: {
      'Content-Type': 'multipart/form-data'
    }
  })
}

// 获取教师授课的课程
export const getMyTeacherCourses = () => {
  return request.get<CourseVO[]>('/course/my-teacher')
}

// 获取学生班级课程
export const getMyStudentCourses = () => {
  return request.get<CourseVO[]>('/course/my-student')
}

// 根据班级ID获取课程列表
export const getCoursesByClassId = (classId: number) => {
  return request.get<CourseVO[]>(`/course/by-class/${classId}`)
}

export const batchUpdateCourseStatus = (data: { ids: number[]; status: number }) => {
  return request.put('/course/batch/status', data)
}

export const batchDeleteCourses = (ids: number[]) => {
  return request.post('/course/batch-delete', ids)
}

File: backend/src/main/java/com/example/classhelper/entity/Course.java
package com.example.classhelper.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 课程实体类
 * 对应数据库表 sys_course
 */
@Data
@TableName("sys_course")
public class Course {

    /**
     * 课程ID，主键自增
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 课程名称
     */
    @TableField("course_name")
    private String courseName;

    /**
     * 学分
     */
    @TableField("credit")
    private Integer credit;

    /**
     * 课时
     */
    @TableField("course_hours")
    private Integer courseHours;

    /**
     * 授课教师ID
     */
    @TableField("teacher_id")
    private Long teacherId;

    /**
     * 所属班级ID
     */
    @TableField("class_id")
    private Long classId;

    /**
     * 课程描述
     */
    @TableField("description")
    private String description;

    /**
     * 课程封面
     */
    @TableField("cover_url")
    private String coverUrl;

    /**
     * 状态：0-禁用，1-启用
     */
    @TableField("status")
    private Integer status;

    /**
     * 开课时间
     */
    @TableField("start_time")
    private LocalDateTime startTime;

    /**
     * 结课时间
     */
    @TableField("end_time")
    private LocalDateTime endTime;

    /**
     * 创建时间
     */
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 逻辑删除标志：0-未删除，1-已删除
     */
    @TableLogic
    @TableField("is_deleted")
    private Integer isDeleted;

    // ==================== 非数据库字段 ====================

    /**
     * 教师姓名（关联查询）
     */
    @TableField(exist = false)
    private String teacherName;

    /**
     * 班级名称（关联查询）
     */
    @TableField(exist = false)
    private String className;

    /**
     * 班级学生人数
     */
    @TableField(exist = false)
    private Integer studentCount;

    /**
     * 课程作业数
     */
    @TableField(exist = false)
    private Integer assignmentCount;

    /**
     * 进行中的作业数
     */
    @TableField(exist = false)
    private Integer activeAssignmentCount;

    /**
     * 课程平均完成率
     */
    @TableField(exist = false)
    private Integer completionRate;

}
